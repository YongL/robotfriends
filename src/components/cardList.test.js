import { shallow } from 'enzyme';
// import { shallow, mount, render, configure } from 'enzyme';
import React from 'react'; // import this because it is jsx
import CardList from './cardList';


// import Adapter from 'enzyme-adapter-react-16';

// configure({adapter: new Adapter()});

// the test script come with react script /// create-react-app 


// shallow only shollow or render one give component it doesn't care anything inside or children
// inside it
// the document doesn't really give a specific direction of where to 
// to put the configure// 
// under src create setupTests.js 
// https://airbnb.io/enzyme/docs/installation/index.html


// it('render Card', () => expect(shallow(<Card />).length).toBe(1));
const robots = [{
    id: 1,
    name: 'some name',
    email: 'some@email.com'
}];


it('render Card', () => expect(shallow(<CardList robots={robots}/>)).toMatchSnapshot());

describe('display props', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<CardList robots={robots}/>);
    });

    it('has props', () => {
        expect(wrapper.props().name).toEqual(robots.name);
    })
})

// jest do snaphot testing, which snapshot the codes (string copy not image)

