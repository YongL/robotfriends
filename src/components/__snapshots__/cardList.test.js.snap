// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Card 1`] = `
<div>
  <Card
    email="some@email.com"
    id={1}
    key="1"
    name="some name"
  />
</div>
`;
